### PR TITLE
Big bertha: System test handling

### PR DESCRIPTION
**Pull request by ArthurAtangana**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [x] ✅ Test
- [x] 🧑‍💻 Refactor

## Description

- Added idling for the elevator
- more logic for when the elevator should stop
- Loader rework
- Added system test, simplifying the input-file to have a more understandable output
- main is running with our test-input-file.txt in the test/resources folder

This is a decently big rework of some of the components. To review it best, please go through the commit history as I documented and separated them on purpose.

The system test is working: just press play and boom roasted.

## How to test

Run main, The output should demonstrate that:
- Request1: Floor 1 going UP
- The elevator goes to floor 1. stops pick up R1(going to floor 7)
- The elevator starts going up floors 
- Request2: Floor 1 going UP
- Request3: Floor 6 going DOWN
- Request4: Floor 5 going UP
- The elevator is still going up floors
- Reaches floor 5. stops, picks up R4 (going to floor 6)
- Elevator goes up again, reaches floor 6, stops, unloads R4
- Elevator goes up again, reaches floor 7, stops, unloads R1
- Request5: Floor 3 going DOWN
- Request6: Floor 1 going UP
- Elevator goes down to service R2 on floor 1
- Elevator doesn't pickup anyone on the way down
- Request7: Floor 2 going UP
- Elevator reaches floor 1. picks up R2 (going to Floor 3) and R6 (going to floor 6)
- Elevator goes up
- Elevator stops on floor 2. pickups R7 (going to floor 6)
- elevator goes up .
- Elevator stops on floor 3, unloads R2
- Elevator goes up
- Elevator stops on floor 6, unloads R6 and R7
- Elevator loads R3 (going to floor 2)
- Elevator goes down
- Elevator stops on floor 3, pickups R5 (going to floor 1)
- Elevator goes down
- Elevator stops on floor 2, unloads R3
- Elevator goes down
- Elevator stops on floor 1, unloads R5
- Elevator Idles.

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: please replace this line with details on why tests have not been included
- [ ] I need help with writing tests

### CODE REVIEW

File: src/Messaging/Receivers/DMA_Receiver.java [hunk 1]
```diff
@@ -43,14 +43,15 @@ public DMA_Receiver(int key) {
      */
     @Override
     public synchronized SystemMessage receive() {
```
Comment (orselane) [inline src/Messaging/Receivers/DMA_Receiver.java]: Remove synchronized keyword (already synchronized on msgBuf inside)

File: src/SchedulerSubsystem/Loader.java [hunk 2]
```diff
@@ -10,32 +12,37 @@
...
     }
-
     /**
      * Runs this operation.
      */
     @Override
     public void run() {
         // Send command to get passengers from floor
-        txFloor.send(new SendPassengersCommand(elevatorState.currentFloor(), elevatorState.direction(), txThis));
+        //System.out.println("loader: ask passengers from the floor");
```
Comment (orselane) [inline src/SchedulerSubsystem/Loader.java]: Remove commented out code (check prints in all changed files please.)

File: src/SchedulerSubsystem/Scheduler.java [hunk 3]
```diff
@@ -8,17 +9,15 @@
...
 import org.junit.experimental.runners.Enclosed;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.runner.RunWith;
 
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (orselane) [inline src/SchedulerSubsystem/Scheduler.java]: Auto-import should only import the needed classes, why is this a wildcard?

File: src/SchedulerSubsystem/Scheduler.java [hunk 4]
```diff
@@ -66,8 +70,33 @@ private boolean isElevatorStopping(ElevatorStateEvent e) {
         union.addAll(floorRequestsToTime.keySet());
         union.addAll(e.passengerCountMap().keySet());
 
+        if (e.passengerCountMap().isEmpty() && getElevatorDirection(e) != getOldestFloorRequest().direction()){
```
Comment (orselane) [inline src/SchedulerSubsystem/Scheduler.java]: This condition should either be a named boolean private method, or commented

**Review by orselane — CHANGES_REQUESTED** [review]

**Comment by orselane** [comment]
Also, there is no system test file. This PR does not modify the system test file, if the PR has a "Yes" for test, shouldn't we have something in the How to Test section?

**Review by orselane — APPROVED** [review]

**Review by mpdesantis — APPROVED** [review]
You busy guy. Looks really nice, and I like the test scenarios. :)

**Pull request merged**